### PR TITLE
Add Node v0.12.x support

### PR DESCRIPTION
**Pull request by MadLittleMods**
Add Node v0.12.x support.

Fixes KABA-CCEAC/node-usb-detection#6

 - Upgraded all native addon C++ calls.
 - Used [nan](https://www.npmjs.com/package/nan)

---

You can depend on it now until the PR gets merged.

```
npm install git://github.com/MadLittleMods/node-usb-detection.git#add-node-v0.12.x-support --save
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-  "version": "1.0.3",
+  "version": "1.1.0",
   "description": "List USB devices in system and detect changes on them.",
   "main": "index.js",
   "dependencies": {
     "bindings": "1.1.0",
-    "eventemitter2": ">=0.4.11"
+    "eventemitter2": ">=0.4.11",
+    "nan": "^1.8.4",
+    "nav": "0.0.2"
```
Comment (adrai) [inline package.json]: why nav?

**Comment by adrai** [comment]
Wow seams nice...
Did you test on windows, linux and mac os x?

File: package.json (hunk 1 above)
Comment (MadLittleMods) [inline package.json]: @adrai Sorry, probably mis-installed that when installing `nan`

**Comment by MadLittleMods** [comment]
@mscdex Only on Windows 8.1 :worried:

**Comment by adrai** [comment]
for the moment I merged to the nan branch

**Comment by MadLittleMods** [comment]
@adrai I removed the extraneous `nav` package.

**Comment by adrai** [comment]
ok tested on os x...

**Comment by MadLittleMods** [comment]
@adrai Make a release and publish to npm please :)

**Comment by adrai** [comment]
done ;-)

**Comment by adrai** [comment]
You're great!

**Comment by adrai** [comment]
Are you interested in taking the ownership of this module?

**Comment by MadLittleMods** [comment]
@adrai I am interested :grinning: - Awesome package!

What does this entail?

**Comment by adrai** [comment]
It's just an idea... I want to discuss this with my boss... and after that I come back to you, ok?

**Comment by MadLittleMods** [comment]
@adrai Sounds good, I look forward discussing the details soon.

**Comment by masnatacion** [comment]
Windows 8.1
Windows Software Development Kit (SDK)
Python 2.7
.NET Framework 2.0, 3.0 and 4.0

I get it this :

MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe".
 To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visua
l Studio 2005 or 3) add the location of the component to the system path if it
is installed elsewhere.  [E:\GitHub\jquery-mobile\node_modules\jquery\node_modu
les\contextify\build\binding.sln]

**Comment by MadLittleMods** [comment]
@masnatacion I had no problem with Visual Studio 2013 Community edition.

**Comment by masnatacion** [comment]
@MadLittleMods your rocks!!! working now :+1:

**Comment by masnatacion** [comment]
@MadLittleMods do u know how can I get device path? 

I only have:

locationId: 337641472,
  vendorId: 6309,
  productId: 770,
  deviceName: 'STORE N GO',
  manufacturer: 'Verbatim',
  serialNumber: 'C400446A92C22022',
  deviceAddress: 11

**Comment by MadLittleMods** [comment]
@masnatacion Create a new issue for your problem.

**Comment by adrai** [comment]
@MadLittleMods next week I should get an answer...

**Comment by adrai** [comment]
@MadLittleMods hangouts?

**Pull request merged**